src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -71,7 +71,7 @@ public override async Task ProvideCompletionsAsync(CompletionContext completionC
                 using var _ = ArrayBuilder<(string name, SymbolKind kind)>.GetInstance(out var result);
 
                 // Suggest names from existing overloads.
-                if (nameInfo.PossibleSymbolKinds.Any(k => k.SymbolKind == SymbolKind.Parameter))
+                if (!nameInfo.PossibleSymbolKinds.IsDefaultOrEmpty && nameInfo.PossibleSymbolKinds.Any(k => k.SymbolKind == SymbolKind.Parameter))
```

Comment:
❗ We should instead required the constructor to initialize `PossibleSyntaxKinds` to a non-default value.

Reply:
Fixed